lib/phoenix/router/helpers.ex

```diff
@@ -223,7 +240,8 @@ defmodule Phoenix.Router.Helpers do
       end
 
       def unquote(:"#{helper}_path")(conn_or_endpoint, unquote(opts), unquote_splicing(vars), params) do
-        path(conn_or_endpoint, segments(unquote(segs), params, unquote(bins)))
+        path(conn_or_endpoint, segments!(unquote(segs), params, unquote(bins),
+              {unquote(helper), unquote(opts), length(unquote(vars))}))
```

Comment:
Hrm, I think we should rather do `unquote(length(vars))` so the work happens at compile time... but now I think I know an approach we can do to keep the variable names without doing the work at runtime time. What if you unquoted this: `unquote(Enum.map(vars, &Macro.to_string/1))`?

This way you get the variable names, and not the arguments, which will improve the error message but still be short compile-time wise.

Reply:
I am assuming those `vars` are rather decent though and not just generated var1/var2 stuff.